assembly/src/assembler/context.rs

```diff
@@ -341,18 +369,23 @@ impl ModuleContext {
             compiled_procs: Vec::new(),
             path: LibraryPath::exec_path(),
             callset: CallSet::default(),
+            proc_map: BTreeMap::new(),
         }
     }
 
     /// Returns a new [ModuleContext] instantiated for compiling library modules.
     ///
     /// A library module must be identified by a unique module path.
-    pub fn for_module(module_path: &LibraryPath) -> Self {
+    pub fn for_module(
+        module_path: &LibraryPath,
+        proc_map: BTreeMap<ProcedureId, ProcedureName>,
+    ) -> Self {
```

Comment:
We probably need to do something similar for the `for_program` constructor above.